src/transforms/operations.js

```diff
@@ -286,6 +286,43 @@ Transforms.removeTextOperation = (transform, path, offset, length) => {
 }
 
 /**
+ * Merge `properties` into state `data`.
+ * `properties` can be an object or an array of `[key, value]` pairs.
+ * The array form can be used to store data with keys different from strings.
```

Comment:
Hey @AlbertHilb can we also change this. Since we're trying to keep the `state.data` JSON serializable, we can't allow keys to be anything other than strings. This will keep it consistent with the other places that `Data` is used now. 

I think the idea of using `Symbol`s for private keys is interesting, but I'd like to keep the scope of this addition smaller for now. If we really decide to allow for non-string keys, we can do it in a separate pass and do it everywhere. Thank you!